sql/hive/src/test/scala/org/apache/spark/sql/sources/hadoopFsRelationSuites.scala

```diff
@@ -554,6 +556,21 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils {
       clonedConf.foreach(entry => configuration.set(entry.getKey, entry.getValue))
     }
   }
+
+  test("SPARK-8887: Explicitly define which data types can be used as dynamic partition columns") {
+    val df = Seq(
+      (1, "v1", Date.valueOf("2015-08-10")),
+      (2, "v2", Date.valueOf("2015-08-11")),
+      (3, "v3", Date.valueOf("2015-08-12"))).toDF("a", "b", "c")
+    withTempDir { file =>
+      intercept[AnalysisException] {
+        df.write.format(dataSourceName).partitionBy("c").save(file.getCanonicalPath)
+      }
+    }
+    intercept[AnalysisException] {
+      df.write.format(dataSourceName).partitionBy("c").saveAsTable("t")
+    }
+  }
```

Comment:
We need to update this test case after putting `DateType` into the set of valid types of partition columns.